style: Sync changes from mozilla-central.

emilio opened the pull request:
See each individual commit for details.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/servo/servo/22167)
<!-- Reviewable:end -->

aethanyc committed:
style: Enable CSS column-span in UA stylesheet, and update test expectations.

We use "column-span: all" in ua.css in Part 3. To be able to flip the pref
in individual wpt tests, we need column-span to be always enabled
in UA stylesheets.

Differential Revision: https://phabricator.services.mozilla.com/D5212

emilio committed:
style: Allow ::slotted()::placeholder.

Differential Revision: https://phabricator.services.mozilla.com/D11132

emilio committed:
style: Implement overflow-wrap: anywhere.

Per https://github.com/w3c/csswg-drafts/issues/2682

Differential Revision: https://phabricator.services.mozilla.com/D11328

BorisChiou committed:
style: Percentage values of translate are serialized as percent for computed values.

Basically, we rewrite the type of generics::transform::Translate and its
ToCss to match the spec. Besides, we always serialize Translate by servo,
so we could drop a lot of duplicated code.

Differential Revision: https://phabricator.services.mozilla.com/D11206

emilio committed:
style: Cleanup text-align.

Use derive(FromPrimitive) instead.

Differential Revision: https://phabricator.services.mozilla.com/D11332

emilio committed:
style: Split up push_applicable_declarations.

Introduce RuleCollector, which contains all the state we need during the
cascade, and allows to reuse a bit of code.

Differential Revision: https://phabricator.services.mozilla.com/D11233

Commit:
style: Add -webkit-appearance:progress-bar and make that the default for <progress> for compatibility with other UAs.

Keep our old 'progressbar' as an alias for now, but unship
'progresschunk' by restricting it to UA/chrome sheets only.
Unship 'progresschunk-vertical' by removing it since it's
not used internally for anything.

Bug: 1501506
Reviewed-by: jwatt

Commit:
style: Add -webkit-appearance:meter and make that the default for <meter> for compatibility with other UAs.

Keep our old value 'meterbar' as an alias for now, but unship 'meterchunk'
by restricting it to UA/chrome sheets only.

Bug: 1501483
Reviewed-by: jwatt

emilio committed:
style: Remove dynamic HTML5 atoms.

Co-authored-by: Nicholas Nethercote <nnethercote@mozilla.com>

Differential Revision: https://phabricator.services.mozilla.com/D11035

emilio committed:
layout: Fix servo build.

TextAlign now implements FromPrimitive instead of an ad-hoc method.

emilio committed:
style: Cargo fix + ./mach fmt.

highfive commented:
<img src="http://www.joshmatthews.net/warning.svg" alt="warning" height=20> **Warning** <img src="http://www.joshmatthews.net/warning.svg" alt="warning" height=20>

* These commits modify **unsafe code**. Please review it carefully!
* These commits modify style and layout code, but no tests are modified. Please consider adding a test!

emilio commented:
@bors-servo r+

bors-servo commented:
:pushpin: Commit 155caba has been approved by `emilio`

<!-- @bors-servo r=emilio 155caba59512cf284d6c2b1c118ed3afeb8fdf49 -->

emilio committed:
style: Revert try -> r#try change.

Since we're in an inconsistent state because mako files weren't updated, and
it's really really ugly.

emilio committed:
style: Manually extinguish multi-line use statements.

emilio commented:
@bors-servo r+

bors-servo commented:
:pushpin: Commit bd9c53c has been approved by `emilio`

<!-- @bors-servo r=emilio bd9c53c5da1ae1c24d4f01f00ff61070ba8bc10f -->

bors-servo commented:
:hourglass: Testing commit bd9c53c5da1ae1c24d4f01f00ff61070ba8bc10f with merge 19b4f35de1ec593ed0ddf11dc0e41ef8f191d496...
